hydromt_wflow/wflow.py

```diff
@@ -3105,6 +3347,519 @@ def setup_1dmodel_connection(
+
+        # Read data
+        domestic_raw = self.data_catalog.get_rasterdataset(
+            domestic_fn,
+            geom=self.region,
+            buffer=2,
+            variables=["dom_gross", "dom_net"],
+        )
+        # Increase the buffer if original resolution is provided
+        if domestic_fn_original_res is not None:
+            buffer = np.ceil(domestic_fn_original_res / abs(domestic_raw.raster.res[0]))
+            domestic_raw = self.data_catalog.get_rasterdataset(
+                domestic_fn,
+                geom=self.region,
+                buffer=buffer,
+                variables=["dom_gross", "dom_net"],
+            )
+        # Check if data is time dependent
+        if "time" in domestic_raw.coords:
+            # Check that this is indeed cyclic data
+            if len(domestic_raw.time) in [12, 365, 366]:
+                _cyclic = True
+                domestic_raw["time"] = domestic_raw.time.astype("int32")
+            else:
+                self.logger.error(
+                    "The provided domestic demand data is cyclic but the time "
+                    "dimension does not match the expected length of 12, 365 or 366."
+                )
+
+        # Get population data
+        pop_raw = None
+        if population_fn is not None:
+            pop_raw = self.data_catalog.get_rasterdataset(
+                population_fn,
+                bbox=domestic_raw.raster.bounds,
+                buffer=2,
+            )
+
+        # Compute domestic demand
+        domestic, pop = workflows.demand.domestic(
+            domestic_raw,
+            ds_like=self.grid,
+            popu=pop_raw,
+            original_res=domestic_fn_original_res,
+        )
+        # Add to grid
+        rmdict = {k: self._MAPS.get(k, k) for k in domestic.data_vars}
+        self.set_grid(domestic.rename(rmdict))
+        if population_fn is not None:
+            self.set_grid(pop, name="population")
+
+        # Update toml
+        if _cyclic and self.get_config("input.cyclic") is None:
+            self.set_config("input.cyclic", [])
+        self.set_config("model.water_demand.domestic", True)
+
+        for demand_type in ["gross", "net"]:
+            self.set_config(
+                f"input.vertical.domestic.demand_{demand_type}",
+                f"domestic_{demand_type}",
+            )
+            if _cyclic:
+                self.config["input"]["cyclic"].append(
+                    f"vertical.domestic.demand_{demand_type}"
+                )
+
+    def setup_other_demand(
+        self,
+        demand_fn: Union[str, dict, xr.Dataset],
+        variables: list = ["ind_gross", "ind_net", "lsk_gross", "lsk_net"],
+        resampling_method: str = "average",
+    ):
+        """Create water demand maps from other sources (e.g. industry, livestock).
+
+        These maps are created from a supplied dataset that either contains one
+        or all of the following variables:
+        - `Domestic` water demand
```

Comment:
Perhaps best to remove this mention of domestic here, since we have a dedicated function for that. Or maybe put it lower on the list/between brackets?